auth/cache.go

```diff
@@ -0,0 +1,66 @@
+package auth
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/manicminer/hamilton/auth"
+)
+
+type CachedAutorestAuthorizer struct {
```

Comment:
how is this cached? should this be HamiltonAutorestAuthorizer?

Reply:
Good catch thanks, it isn't doing any caching (that functionality remains in the origin struct in the main hamilton auth package)

Reply:
Going to rename all the types to drop the Autorest prefix too since they now live in their own package